### PR TITLE
feat: support more sql conditions on string/number type

### PR DESCRIPTION
**Pull request by llmin**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary

(describe what this merge request does)

## Implementation highlights

(describe how the merge request does for feature changes, share the RFC link if it has)

## Test checklist

- [x] add new test cases
- [ ] all code changes are covered by unit tests
- [ ] end-to-end tests
  - [ ] deploy web console with CloudFront + S3 + API gateway
  - [ ] deploy web console within VPC
  - [ ] deploy ingestion server
    - [ ] with MSK sink
    - [ ] with KDS sink
    - [ ] with S3 sink
  - [ ] deploy data processing
  - [ ] deploy data modeling
    - [ ] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [ ] deploy with reporting

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

## Miscellaneous

- [ ] introduce new symbol link source file(s) to be shared among infra code, web console frontend, and web console backend

### CODE REVIEW

File: src/control-plane/backend/lambda/api/service/quicksight/reporting-utils.ts [hunk 1]
```diff
@@ -507,6 +507,23 @@ export function getFunnelTableVisualDef(visualId: string, viewName: string, even
             DataSetIdentifier: viewName,
             ColumnName: 'rate',
           },
+          FormatConfiguration: {
+            FormatConfiguration: {
```
Comment (zxkane) [inline src/control-plane/backend/lambda/api/service/quicksight/reporting-utils.ts:511]: why has the duplicated field name?

File: src/control-plane/backend/lambda/api/test/api/sql-builder.test.ts [hunk 2]
```diff
@@ -1484,6 +1484,11 @@ describe('SQL Builder test', () => {
 
   test('funnel table visual sql - conditions', () => {
 
+
+    console.log([10, 20, 30].join(','));
```
Comment (zxkane) [inline src/control-plane/backend/lambda/api/test/api/sql-builder.test.ts:1488]: remove those test code

File: src/control-plane/backend/lambda/api/service/quicksight/sql-builder.ts [hunk 3]
```diff
@@ -13,12 +13,13 @@
...
 import { format } from 'sql-formatter';
 import { ExploreComputeMethod, ExploreConversionIntervalType, ExploreGroupColumn, ExplorePathNodeType, ExplorePathSessionDef, ExploreRelativeTimeUnit, ExploreTimeScopeType, MetadataPlatform, MetadataValueType } from '../../common/explore-types';
+import { logger } from '../../common/powertools';
 
 export interface Condition {
   readonly category: 'user' | 'event' | 'device' | 'geo' | 'app_info' | 'traffic_source' | 'other';
   readonly property: string;
   readonly operator: string;
-  readonly value: string;
+  readonly value: any[];
```
Comment (zxkane) [inline src/control-plane/backend/lambda/api/service/quicksight/sql-builder.ts:22]: add test cases for the value has more than one item

**Review by zxkane — CHANGES_REQUESTED** [review]

File: src/control-plane/backend/lambda/api/service/quicksight/sql-builder.ts (hunk 3 above)
Comment (llmin) [reply src/control-plane/backend/lambda/api/service/quicksight/sql-builder.ts:22]: covered by "condition sql" test case

File: src/control-plane/backend/lambda/api/test/api/sql-builder.test.ts (hunk 2 above)
Comment (llmin) [reply src/control-plane/backend/lambda/api/test/api/sql-builder.test.ts]: removed

File: src/control-plane/backend/lambda/api/service/quicksight/sql-builder.ts [hunk 4]
```diff
@@ -13,12 +13,13 @@
 
 import { format } from 'sql-formatter';
 import { ExploreComputeMethod, ExploreConversionIntervalType, ExploreGroupColumn, ExplorePathNodeType, ExplorePathSessionDef, ExploreRelativeTimeUnit, ExploreTimeScopeType, MetadataPlatform, MetadataValueType } from '../../common/explore-types';
+import { logger } from '../../common/powertools';
 
 export interface Condition {
   readonly category: 'user' | 'event' | 'device' | 'geo' | 'app_info' | 'traffic_source' | 'other';
   readonly property: string;
```
Comment (tyyzqmf) [inline src/control-plane/backend/lambda/api/service/quicksight/sql-builder.ts:20]: How to get `category` of `property` ?

File: src/control-plane/backend/lambda/api/service/quicksight/reporting-utils.ts (hunk 1 above)
Comment (llmin) [reply src/control-plane/backend/lambda/api/service/quicksight/reporting-utils.ts:511]: Although it looks strange, it is correct.

**Pull request merged**